### PR TITLE
Fix GIF global color table flag check

### PR DESCRIPTION
**Pull request by nbianca**
Zero is a truthy value, which was a problem when checking if specific bit is set.

### CODE REVIEW

**Comment by sdsykes** [comment]
Yeh, I'll merge this.

Do you have any test images that exercise this code path? (And other code paths in animated? for that matter).

**Comment by nbianca** [comment]
That check used to be always true, so I just added an animated GIF without a GCT.

File: lib/fastimage.rb [hunk 1]
```diff
@@ -585,7 +585,7 @@ def animated?
 
       # fields (1) + bg color (1) + pixel ratio (1)
       fields = @stream.read(3).unpack("CCC")[0]
-      if fields & 0x80 # Global Color Table
+      if fields & 0x80 != 0 # Global Color Table
```
Comment (Nakilon) [inline lib/fastimage.rb:588]: Or maybe
```
if fields[10].zero?
```

Comment (nbianca) [reply lib/fastimage.rb:588]: I think that would have worked if it were `fields[7]`, but I do not think that makes the code more readable.

**Pull request merged**